python/ray/tests/test_batch_node_provider_unit.py

```diff
@@ -415,6 +419,70 @@ def test_terminate_safeguards():
     assert len(nodes) == 1
 
 
+@pytest.mark.skipif(sys.platform.startswith("win"), reason="Not relevant on Windows.")
+def test_terminate_node_in_multihost_replica():
+    """Test multi-host replica deletion logic for KubeRay.
+
+    Tests manually deleting a node in a multi-host replica
+    and verifying that the entire replica is scaled down.
+    Nodes belonging to the same multi-host replica are identified
+    through a replicaIndex label set by a GKE webhook.
+    """
+    # create 4 TPU nodes with MockBatchingNodeProvider
+    node_provider = MockBatchingNodeProvider(
+        provider_config={
+            DISABLE_LAUNCH_CONFIG_CHECK_KEY: True,
+            DISABLE_NODE_UPDATERS_KEY: True,
+            FOREGROUND_NODE_LAUNCH_KEY: True,
+        },
+        cluster_name="test-cluster",
+        _allow_multiple=True,
+    )
+
+    num_tpu_workers = 4
+    for i in range(num_tpu_workers):
+        node_provider._add_node(node_type="TPU", node_kind=NODE_KIND_WORKER)
+
+    # Set replica_index in node_data for all workers
+    workers = node_provider.non_terminated_nodes(
+        tag_filters={TAG_RAY_NODE_KIND: NODE_KIND_WORKER}
+    )
+    assert len(workers) == num_tpu_workers
+    index = 0
+    for node_id in workers:
```

Comment:
Maybe we don't need `index` here. Just use

```
for .... enumerate(workers):
```

Reply:
302b69a